lib/testers/rule-tester.js

```diff
@@ -396,10 +396,10 @@ class RuleTester {
          * @private
          */
         function assertASTDidntChange(beforeAST, afterAST) {
-            if (!lodash.isEqual(beforeAST, afterAST)) {
 
-                // Not using directly to avoid performance problem in node 6.1.0. See #6111
-                assert.deepStrictEqual(beforeAST, afterAST, "Rule should not modify AST.");
+            // Feature detect the Node.js implementation and use that if available.
+            if (util.isDeepStrictEqual && !util.isDeepStrictEqual(beforeAST, afterAST) || !lodash.isEqual(beforeAST, afterAST)) {
```

Comment:
Could the first two conditions be parenthesized, just to avoid any ambiguity for future readers? (Yes, I know the order of operations is clear here, but just to be absolutely safe.)

```js
if ((util.isDeepStrictEqual && !util.isDeepStrictEqual(beforeAST, afterAST)) || !lodash.isEqual(beforeAST, afterAST)) {
```

Thanks!

Reply:
Addressed.